charts/tidb-lightning/templates/job.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "tidb-lightning.name" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "tidb-lightning.name" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: tidb-lightning
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
+spec:
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "tidb-lightning.name" . }}
+        app.kubernetes.io/instance:  {{ .Release.Name }}
+        app.kubernetes.io/component: tidb-lightning
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/path: "/metrics"
+        prometheus.io/port: "8289"
+    spec:
+      {{ if .Values.dataSource.local.nodeName -}}
+      nodeName: {{ .Values.dataSource.local.nodeName }}
+      {{ else -}}
+      initContainers:
+      - name: data-retriever
+        image: {{ .Values.dataSource.remote.rcloneImage }}
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -euo pipefail
+          filename=$(basename {{ .Values.dataSource.remote.path }})
+          rclone --config /etc/rclone/rclone.conf copy -P {{ .Values.dataSource.remote.path }} /data
+          cd /data && tar xzvf ${filename}
+        volumeMounts:
+        - name: credentials
+          mountPath: /etc/rclone
+        - name: data
+          mountPath: /data
+      {{ end -}}
+      restartPolicy: Never      # if lightning fails, manual intervention is required so no restart
+      containers:
+      - name: tidb-lightning
+        image: {{ .Values.image }}
+        imagePullPolicy: {{ .Values.imagePullPolicy | default "IfNotPresent" }}
+        command:
+        - /bin/sh
+        - -c
+        - |
+          data_dir=$(dirname $(find /var/lib/tidb-lightning -name metadata 2>/dev/null) 2>/dev/null)
```

Comment:
Why would `dirname` need `2>/dev/null` ?
We should always have `set -euo pipefail` for our shell scripts.
Although it seems this is only a script to find the metadata and I think we should already know where it is?

Reply:
The remote data is archived from a tarball file, and the extracted directory is unknown.

Reply:
Why is the directory unknown? Is it named after data given by the user?

Why would `dirname` need `2>/dev/null` ?

It also looks like this should only be ran if certain options are set below.

Reply:
We don't know if the tarball filename is `dump_dir.tar.gz`, because the backup tarball file is created by users, they can name it whatever they want. So we can not extract the dir name from the tarball file. Also, the dumped files may be broken or even not mydumper files. So this line may fail in different ways.